e2e_playwright/st_dialog_test.py

```diff
@@ -411,6 +416,58 @@ def test_dialog_with_dataframe_shows_toolbar(
     assert_snapshot(df_toolbar, name="st_dialog-shows_full_dataframe_toolbar")
 
 
+def test_dialog_with_rerun_closes_even_if_button_is_clicked_multiple_times(app: Page):
```

Comment:
I am myself unsure about whether we want to have the test. The test might succeed even without the fix but only fail sometimes, so its not a 100% guarantee catch. But it might still be a good indicator to become aware of the issue eventually if the test starts to fail / fails from time to time.

Reply:
Probably better than nothing 👍